### PR TITLE
initial dockerfile

### PR DESCRIPTION
**Pull request by danasaur**
to test __init__.py I had to make a copy outside of the folder process_pose_data/

The imports specify:

from process_pose_data.honeycomb_io import *

I think I could run the __init__.py from inside the process_pose_data folder if I changed the imports to:

from honeycomb_io import *

I committed the copy and left the original __init__.py because I'm not sure how that's currently being used!

### CODE REVIEW

**Comment by tcquinn** [comment]
A few thoughts/questions:
* The way I usually install `wf-process-pose-data` and its dependencies is simply `pip install wf-process-pose-data`(or `pip install PATH_TO_PACKAGE_DIRECTORY` if installing from local files). Then `pip` follows the dependency tree, finds all of the dependencies (and subdependencies), and installs everything automatically. It looks like you are installing all of the the dependencies (and subdependencies) "manually" (either by listing each of them in a `requirements.txt` file or with a separate `RUN` command for each one), but you're _not_ ever installing `wf-process-pose-data`. There may be a good reason for this, but my first instinct is that it's unnecessarily complex, a bit fragile (the list of dependencies in `requirements.txt` would need to be updated every time any package in the tree changes its dependencies), and may be leading to some of the other strange behavior you noticed.
* As you noticed, the `import` statements in my module-level `__init__.py` (`./process_pose_data/__init__.py`) reference the submodules using the full package path (e.g., `from process_pose_data.honeycomb_io import *`) rather than the local directory path. My guess is that this is generating errors for you because of the above (i.e., you're never actually installing the package `wf-process-pose-data`). I'm happy to rewrite these references, but based on my current (possibly incorrect) understanding, I don't think it's necessary.
* Relatedly, I'm not sure why you're explicitly running `__init__.py`. As I've set this up, the `__init__.py` file is just there to declare the structure of the namespace. It's not set up as a user executable. Again, there could be a good reason for this, but it's not a pattern I'm familiar with.
* Finally, I'm not sure why you created _another_ `__init__.py` at the _package_ level (i.e., the root level of the repo). I'm not used to seeing one there, since the module is down in the `process_pose_data` directory.

The caveat to all of this is that I'm not a Docker expert so this all may just reflect my ignorance of different Docker usage patterns. @optimuspaul may be able to contribute some wisdom in this regard, since he's been our Docker mastermind.

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+ENV DEBIAN_FRONTEND=noninteractive
+
+RUN apt-get install -y python3 python3-pip apt-utils libgl1-mesa-glx libglib2.0-0
+
+WORKDIR /wf-process-pose-data
+
+COPY . .
+
+RUN yes | pip3 install -r requirements.txt
```
Comment (optimuspaul) [inline Dockerfile:13]: `RUN pip install -e .`

File: __init__.py [hunk 2]
```diff
@@ -0,0 +1,10 @@
+from process_pose_data.honeycomb_io import *
```
Comment (optimuspaul) [inline __init__.py:1]: We don't need this file

File: requirements.txt [hunk 3]
```diff
@@ -0,0 +1,15 @@
+wf-pose-tracking-3d>=0.1.0
```
Comment (optimuspaul) [inline requirements.txt:1]: We don't need this file

File: Dockerfile (hunk 1 above)
Comment (optimuspaul) [reply Dockerfile:13]: this will install all the dependencies and make this package available to python.

**Comment by danasaur** [comment]
I've never worked with setup.py / init files and just didn't realize this package was pip installable. This is much cleaner than using the requirements.txt ! 

I only ran the init to test that the imports in it would work. I had to add three things to get that file to run:

1. Dockerfile line 7: libgl1-mesa-glx libglib2.0-0
2. setup.py line 22: 'attrs>=19.3.0'

Screen Shot 2020-08-18 at 9 08 35 PM
Screen Shot 2020-08-18 at 9 01 46 PM

Is there a script somewhere that executes these libraries? I could test running that in this container.

**Comment by tcquinn** [comment]
Good catch on the `attrs` dependency. Looks like that snuck in to the `wf-cv-utils` library and never got documented in the `setup.py` for that library It should actually be added there and then we can remove it here. Do you want to just submit a PR for that?

Looking at the `libgl1-mesa-glx libglib2.0-0` dependencies now.

**Comment by optimuspaul** [comment]
`attrs` is probably my fault. I had introduced that in some code I contributed oopsies

**Comment by tcquinn** [comment]
Looks like `libgl1-mesa-glx` and `libglib2.0-0` are indeed both installed in my Ubuntu setup. I suspect that they are required for OpenCV, but I had already installed them for some other reason, and therefore never got the error.

**Comment by tcquinn** [comment]
Agree that we should at least have a test script in the repo. One of my tasks for the next week or so is to write a proper worker so we can integrate all of this stuff in our pipeline. When I do that, I can also make a test script.

**Comment by tcquinn** [comment]
If you're up for it, @danasaur , I think the next task here is to fiddle with the `wf-cv-utils` set up. One of the original inspirations for Dockerizing this universe was that the default Python OpenCV installation (`pip install opencv-python`) lacks support for the H264 video codec that our videos use because the latter is not fully open source and the creator of the `opencv-python` package wanted to keep his package free of such licensing complexities. More specifically, I think the issue is that `opencv-python` contains its own version of `ffmpeg` which is compiled without support for the codec. @optimuspaul had suggested that we might figure out the right install sequence to get H264 support (install `ffmpeg` from source with the right options, then install OpenCV from source and point it at this version of `ffmpeg`, or whatever) and then capture that install sequence in a Dockerfile/image for `wf-cv-utils`. This will take some Googling and tinkering. Let us know if that's something you'd be game to tackle.

**Comment by tcquinn** [comment]
More here on the H264 issue:
https://github.com/skvark/opencv-python/issues/100
https://github.com/skvark/opencv-python/issues/207
https://github.com/skvark/opencv-python/issues/299
https://github.com/skvark/opencv-python/issues/326

**Comment by danasaur** [comment]
cleaning up open PRs

**Pull request closed without merging**